google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/apis/CloudBomInspection.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.apis;
+
+import com.google.cloud.tools.libraries.json.CloudLibrary;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.psi.xml.XmlTag;
+import com.intellij.util.IncorrectOperationException;
+import com.intellij.util.xml.GenericDomValue;
+import com.intellij.util.xml.highlighting.DomElementsInspection;
+import java.util.Set;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.idea.maven.dom.model.MavenDomDependency;
+import org.jetbrains.idea.maven.dom.model.MavenDomProjectModel;
+
+/**
+ * Base {@link DomElementsInspection} for inspecting google-cloud-java BOM related issues in the
+ * project's pom.xml.
+ */
+class CloudBomInspection extends DomElementsInspection<MavenDomProjectModel> {
```

Comment:
Should we make it `abstract` to note this is not real implementation?

Reply:
yes good idea